src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -179,13 +181,17 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
 
                     var selectedPlatform = PlatformNegotiation.GetNearestPlatform(overridePlatformNegotiationMetadataValue, projectInstance.GetPropertyValue(PlatformMetadataName), projectInstance.GetPropertyValue(PlatformsMetadataName), projectInstance.GetPropertyValue(PlatformLookupTableMetadataName), requesterInstance.GetPropertyValue(PlatformLookupTableMetadataName), projectInstance.FullPath, requesterInstance.GetPropertyValue(PlatformMetadataName));
 
-                    if (selectedPlatform.Equals(String.Empty))
+                    if (enableDynamicPlatformFiltering && !selectedPlatform.Item1)
+                    {
+                        continue;
```

Comment:
Project being removed from graph might deserve low-priority log message for clarity during troubleshooting.